sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/AsyncVoidMethod.cs

```diff
@@ -66,7 +77,8 @@ protected override void Initialize(SonarAnalysisContext context)
 
         private static bool IsExceptionToTheRule(MethodDeclarationSyntax methodDeclaration, IMethodSymbol methodSymbol) =>
             methodSymbol.IsEventHandler() ||
-            IsUsedAsEventHandler(methodDeclaration);
+            IsUsedAsEventHandler(methodDeclaration) ||
+            HasAnyMsTestV1AllowedAttribute(methodSymbol);

```

Comment:
This could probably would have been simpler if it was the other way around - report only on methods with attributes coming from the assembly of MSTest V2. Yes, it will not be future proof, because some day MS could release V3, but I think it would be far in the future... Anyway, this is just a note and I don't think the implementation should be changed if you have doubts.